osbs/api.py

```diff
@@ -305,7 +302,8 @@ def _create_scratch_build(self, build_request):
             builder_img['kind'] = 'DockerImage'
             builder_img['name'] = ref
 
-        build_config_name = 'scratch-%s' % datetime.datetime.now().strftime('%Y%m%d%H%M%S')
+        build_config_name = 'scratch-%s' % (
+            build_json['spec']['output']['to']['name'][-20:])
```

Comment:
`spec.output.to.name` refers to the image that this build should produce. This seems like an attempt to re-use the timestamp and random string used there. I like that, but it's not very clear from the code and it seems somewhat brittle.

How about something like this:
```python
output_image_name = build_json['spec']['output']['to']['name']
# Reuse random string and timestamp values.
build_config_name = 'scratch-%s-%s' % tuple(output_image_name.rsplit('-', 2)[-2:])
```